operator/internal/validation/lokistack.go

```diff
@@ -115,6 +120,89 @@ func (v *LokiStackValidator) validateHashRingSpec(s lokiv1.LokiStackSpec) field.
 	return nil
 }
 
+func (v *LokiStackValidator) validateLimitsSpec(s *lokiv1.LimitsSpec) field.ErrorList {
+	if s == nil {
+		return nil
+	}
+
+	if s.Global == nil && s.Tenants == nil {
+		return nil
+	}
+
+	var errors field.ErrorList
+	if s.Global != nil && s.Global.IngestionLimits != nil {
+		fp := field.NewPath("spec", "limits", "global", "ingestion", "desiredRate")
+		errors = append(errors, validateIngestionLimits(s.Global.IngestionLimits, fp)...)
+	}
+
+	for id, tenant := range s.Tenants {
+		if tenant.IngestionLimits == nil {
+			continue
+		}
+
+		fp := field.NewPath("spec", "limits", "tenants", id, "ingestion", "desiredRate")
```

Comment:
Question, why do we have in this function the field path lead to `desiredRate` but then in `validateTenantIngestionLimits` we use on the second set of if's `validateTenantIngestionLimits`? Shouldn't we be consistent and always use the same? Especially because it's the same error. I would always go with `perStreamRateLimit` since that is the oldest field.

Reply:
I have simplifed the validation funcs a bit, PTAL